src/representation_theory/basic.lean

```diff
@@ -298,6 +302,33 @@ local notation ρV ` ⊗ ` ρW := tprod ρV ρW
 @[simp]
 lemma tprod_apply (g : G) : (ρV ⊗ ρW) g = tensor_product.map (ρV g) (ρW g) := rfl
 
+lemma tprod_one_as_module {V W : Type*} [add_comm_group V] [add_comm_group W] [module k V]
+  [module k W] (ρ : representation k G V) (r : monoid_algebra k G) (x : V)
+  (y : W) : (r • (x ⊗ₜ y : (ρ.tprod 1).as_module) : (ρ.tprod 1).as_module)
+    = (r • x : ρ.as_module) ⊗ₜ y :=
+begin
+  show as_algebra_hom _ _ _ = as_algebra_hom _ _ _ ⊗ₜ _,
+  simp only [as_algebra_hom_def, monoid_algebra.lift_apply,
+    tprod_apply, monoid_hom.one_apply, linear_map.finsupp_sum_apply,
+    linear_map.smul_apply, tensor_product.map_tmul, linear_map.one_apply],
+  erw tensor_product.sum_tmul,
+  refl,
+end
+
+lemma one_tprod_as_module {V W : Type*} [add_comm_group V] [add_comm_group W] [module k V]
+  [module k W] (ρ : representation k G W) (r : monoid_algebra k G) (x : V)
+  (y : W) : (r • (x ⊗ₜ y : (tprod 1 ρ).as_module) : (tprod 1 ρ).as_module)
+    = x ⊗ₜ (r • y : ρ.as_module) :=
+begin
+  show as_algebra_hom _ _ _ = _ ⊗ₜ as_algebra_hom _ _ _,
+  simp only [as_algebra_hom_def, monoid_algebra.lift_apply,
+    tprod_apply, monoid_hom.one_apply, linear_map.finsupp_sum_apply,
+    linear_map.smul_apply, tensor_product.map_tmul, linear_map.one_apply],
+  erw tensor_product.tmul_sum,
+  simp only [tensor_product.tmul_smul],
+  refl,
+end
```

Comment:
I have a few remarks here:

- We already have `V`, `W` and several other arguments in `variables` so let's use these instead.
- I think it is clearer to omit some of these type ascriptions.
- I think we should also add the corresponding "right-handed" `tprod_one` lemma.
- I would prefer if we could eliminate the `erw`. Maybe it's OK but do we really need to lean this hard on `defeq`ness?

Anyway, here's an outline of what I have in mind:
```suggestion
lemma smul_tprod_one_as_module (r : monoid_algebra k G) (x : V) (y : W) :
  (r • (x ⊗ₜ y) : (ρV.tprod 1).as_module) = (r • x : ρV.as_module) ⊗ₜ y :=
begin
  show as_algebra_hom _ _ _ = as_algebra_hom _ _ _ ⊗ₜ _,
  simp only [as_algebra_hom_def, monoid_algebra.lift_apply,
    tprod_apply, monoid_hom.one_apply, linear_map.finsupp_sum_apply,
    linear_map.smul_apply, tensor_product.map_tmul, linear_map.one_apply],
  erw tensor_product.sum_tmul,
  refl,
end

lemma smul_one_tprod_as_module (r : monoid_algebra k G) (x : V) (y : W) :
  (r • (x ⊗ₜ y) : ((1 : representation k G V).tprod ρW).as_module) = x ⊗ₜ (r • y : ρW.as_module) :=
begin
  show as_algebra_hom _ _ _ = _ ⊗ₜ as_algebra_hom _ _ _,
  simp only [as_algebra_hom_def, monoid_algebra.lift_apply,
    tprod_apply, monoid_hom.one_apply, linear_map.finsupp_sum_apply,
    linear_map.smul_apply, tensor_product.map_tmul, linear_map.one_apply],
  erw tensor_product.tmul_sum,
  simp only [tensor_product.tmul_smul],
  refl,
end
```